File: samples/text/basic/python/ai_studio.py
"""Basic text generation against the Gemini Developer API (AI Studio).

Surface: AI Studio (api-key authenticated).
SDK:     google-genai (unified Python SDK).
Auth:    GEMINI_API_KEY in the environment — picked up by Client() automatically.
"""

from google import genai


def main(model: str = "gemini-3-flash-preview", prompt: str | None = None) -> dict:
    client = genai.Client()

    response = client.models.generate_content(
        model=model,
        contents=prompt or "Explain transformers to a senior backend engineer in three sentences.",
    )

    usage = response.usage_metadata
    return {
        "text": response.text,
        "model": model,
        "finish_reason": _finish_reason(response),
        # Full usage_metadata dump — the host runner extracts modality
        # breakdowns, thinking tokens, cache details from this.
        "usage_metadata": usage.model_dump(mode="json") if usage else None,
    }


def _finish_reason(response) -> str | None:
    candidates = getattr(response, "candidates", None) or []
    if not candidates:
        return None
    fr = getattr(candidates[0], "finish_reason", None)
    return getattr(fr, "name", str(fr)) if fr else None


if __name__ == "__main__":
    import json

    print(json.dumps(main(), indent=2, default=str))

File: samples/text/basic/typescript/ai-studio.ts
/**
 * Basic text generation against the Gemini Developer API (AI Studio).
 *
 * Surface: AI Studio (api-key authenticated).
 * SDK:     @google/genai (unified TypeScript SDK).
 * Auth:    GEMINI_API_KEY in the environment.
 */

import { GoogleGenAI } from '@google/genai'

export async function main(opts: { model?: string; prompt?: string } = {}) {
  const model = opts.model ?? 'gemini-3-flash-preview'
  const prompt =
    opts.prompt ??
    'Explain transformers to a senior backend engineer in three sentences.'

  // GoogleGenAI({}) reads GEMINI_API_KEY from process.env automatically.
  const client = new GoogleGenAI({})

  const response = await client.models.generateContent({
    model,
    contents: prompt,
  })

  const usage = response.usageMetadata
  return {
    text: response.text,
    model,
    usage: {
      promptTokens: usage?.promptTokenCount,
      outputTokens: usage?.candidatesTokenCount,
      thinkingTokens: usage?.thoughtsTokenCount,
      cachedTokens: usage?.cachedContentTokenCount,
      totalTokens: usage?.totalTokenCount,
    },
  }
}

if (require.main === module) {
  main().then((r) => console.log(JSON.stringify(r, null, 2)))
}
